### PR TITLE
[CI] split conda recipes and linting to separate files in azure pipelines

### PR DESCRIPTION
**Pull request by icfaust**
## Description

This follows the conventions already existing in the codebase to simplify the analysis of CI and maintenance of the azure pipelines in sklearnex, this matches conda environment CI jobs in sklearnex.

No performance analysis required.

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

**Performance**

- [x] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [x] I have provided justification why performance has changed or why changes are not expected.
- [x] I have provided justification why quality metrics have changed or why changes are not expected.
- [x] I have extended benchmarking suite and provided corresponding scikit-learn_bench PR if new measurable functionality was introduced in this PR.

### CODE REVIEW

File: .ci/pipeline/linting.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      echo "##vso[task.setvariable variable=ISORT_VERSION]$(grep -zoP '(?<=isort\n    rev: ).*' .pre-commit-config.yaml)"
+      echo "##vso[task.setvariable variable=BLACK_VERSION]$(grep -zoP '(?<=black\n    rev: ).*' .pre-commit-config.yaml)"
+      echo "##vso[task.setvariable variable=PYTHON_LINT_VERSION]$(grep -oP -m 1 '(?<=python).*' .pre-commit-config.yaml)"
+    displayName: 'Collect lint versions'
+  - task: UsePythonVersion@0
+    inputs:
+      versionSpec: $(PYTHON_LINT_VERSION)
+      addToPath: true
+  - script: |
+      python -m pip install --upgrade pip setuptools
```
Comment (david-cortes-intel) [inline .ci/pipeline/linting.yml:27]: Isn't it possible to re-use the pre-commit hook?

Comment (icfaust) [reply .ci/pipeline/linting.yml]: done

File: .ci/pipeline/linting.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#===============================================================================
+steps:
+  - script: |
+      python -m pip install --upgrade pip pre-commit
+      pre-commit install
+      pre-commit run --all-files
```
Comment (david-cortes-intel) [inline .ci/pipeline/linting.yml:20]: Shouldn't this have another command afterwards to ensure that no changes from the current commit happen after the pre-commit is executed? As a test, could you add some change here that would break formatting to see if the CI job reports it as a failure?

**Comment by icfaust** [comment]
image
https://dev.azure.com/daal/daal4py/_build/results?buildId=41529&view=results
@david-cortes-intel

File: .ci/pipeline/linting.yml (hunk 2 above)
Comment (david-cortes-intel) [reply .ci/pipeline/linting.yml:20]: Thanks, I see that it failed correctly and reported the offending file.

Would be ideal to have the diff shown if that's not too much work, but this is good enough.

**Review by david-cortes-intel — APPROVED** [review]
Let's wait for green tickets in the new CI entries now.

**Comment by icfaust** [comment]
@david-cortes-intel step 4 on their website details the approach taken. https://pre-commit.com/  This PR is starting to have feature creep.

**Comment by icfaust** [comment]
@david-cortes-intel thanks for the reviews, It does look now better than it was, especially with the grep hacking gone.
